Source/Particles/Gather/FieldGather.H

```diff
@@ -59,7 +61,15 @@ void doGatherShapeN (const amrex::ParticleReal xp,
                      const amrex::GpuArray<amrex::Real, 3>& dx,
                      const amrex::GpuArray<amrex::Real, 3>& xyzmin,
                      const amrex::Dim3& lo,
-                     const long n_rz_azimuthal_modes)
+                     const long n_rz_azimuthal_modes
+#ifdef PULSAR
+                     ,
+                     amrex::GpuArray<amrex::Real, AMREX_SPACEDIM> const& dom_lo = {},
+                     amrex::GpuArray<amrex::Real, AMREX_SPACEDIM> const& dom_hi = {},
+                     amrex::Real cur_time = 0._rt )
+#else
+                     )
+#endif
```

Comment:
This stuff is used to apply the external E + B field, right? Instead of modifying these function calls, you could define another functor like the ones in `Particles/Gather/GetExternalFields.H`, I think that would be cleaner.

Reply:
Not saying this needs to be done now btw, I know you have a talk to prepare for, but in the long run I think it will isolate the changes made for the pulsar problem in way that makes it easier to keep synced with `dev`.

Reply:
Hi @atmyers . I have moved this to a different file now `Particles/Gather/GatherExternalPulsarFieldOnGrid.H` so that it is lest invasive. Please let me know if you have any corrections or a better way of doing this